django/urls/converters.py

```diff
@@ -0,0 +1,48 @@
+from django.conf import settings
+from django.utils import lru_cache
+
+
+class BaseConverter(object):
+    pass
+
+
+class IntConverter(BaseConverter):
+    regex = '[0-9]+'
```

Comment:
I don't think there isn't a better alternative. Talking about "positive integers" means we still defer thinking about positive and negative ints to the end user. 

```
path('articles/<posint:year>/', views.year_archive),
```

... isn't at all as nice. Since MOST people still won't have to care about negative ints, I still think keeping "int" as the name is a good default, even though not 100% accurate.

Also: there seems to be a common case for negative FLOATS, namely coordinates. But since the DEP says floats shouldn't be included that use-case is nicely sidestepped. 

There's also two nice workarounds to all this: just specify str as type and do the typecasting in the template, or specify your own type that supports negative floats/ints.